plugin/pkg/scheduler/algorithm/priorities/priorities.go

```diff
@@ -68,45 +69,57 @@ func getNonzeroRequests(requests *api.ResourceList) (int64, int64) {
 	} else {
 		out_memory = requests.Memory().Value()
 	}
-	return out_millicpu, out_memory
+
+	if (*requests.NvidiaGPU() == resource.Quantity{}) {
+		out_nvidiaGPU = defaultNvidiaGPURequest
+	} else {
+		out_nvidiaGPU = requests.NvidiaGPU().Value()
+	}
+
+	return out_millicpu, out_memory, out_nvidiaGPU
 }
 
 // Calculate the resource occupancy on a node.  'node' has information about the resources on the node.
 // 'pods' is a list of pods currently scheduled on the node.
 func calculateResourceOccupancy(pod *api.Pod, node api.Node, pods []*api.Pod) schedulerapi.HostPriority {
 	totalMilliCPU := int64(0)
 	totalMemory := int64(0)
+	totalNvidiaGPU := int64(0)
 	capacityMilliCPU := node.Status.Allocatable.Cpu().MilliValue()
 	capacityMemory := node.Status.Allocatable.Memory().Value()
+	capacityNvidiaGPU := node.Status.Allocatable.NvidiaGPU().Value()
 
 	for _, existingPod := range pods {
 		for _, container := range existingPod.Spec.Containers {
-			cpu, memory := getNonzeroRequests(&container.Resources.Requests)
+			cpu, memory, nvidiaGPU := getNonzeroRequests(&container.Resources.Requests)
 			totalMilliCPU += cpu
 			totalMemory += memory
+			totalNvidiaGPU += nvidiaGPU
 		}
 	}
 	// Add the resources requested by the current pod being scheduled.
 	// This also helps differentiate between differently sized, but empty, nodes.
 	for _, container := range pod.Spec.Containers {
-		cpu, memory := getNonzeroRequests(&container.Resources.Requests)
+		cpu, memory, nvidiaGPU := getNonzeroRequests(&container.Resources.Requests)
 		totalMilliCPU += cpu
 		totalMemory += memory
+		totalNvidiaGPU += nvidiaGPU
 	}
 
 	cpuScore := calculateScore(totalMilliCPU, capacityMilliCPU, node.Name)
 	memoryScore := calculateScore(totalMemory, capacityMemory, node.Name)
+	nvidiaGPUScore := calculateScore(totalNvidiaGPU, capacityNvidiaGPU, node.Name)
 	glog.V(10).Infof(
-		"%v -> %v: Least Requested Priority, Absolute/Requested: (%d, %d) / (%d, %d) Score: (%d, %d)",
+		"%v -> %v: Least Requested Priority, Absolute/Requested: (%d, %d, %d) / (%d, %d, %d) Score: (%d, %d, %d)",
 		pod.Name, node.Name,
-		totalMilliCPU, totalMemory,
-		capacityMilliCPU, capacityMemory,
-		cpuScore, memoryScore,
+		totalMilliCPU, totalMemory, totalNvidiaGPU,
+		capacityMilliCPU, capacityMemory, capacityMemory,
+		cpuScore, memoryScore, nvidiaGPUScore,
 	)
 
 	return schedulerapi.HostPriority{
 		Host:  node.Name,
-		Score: int((cpuScore + memoryScore) / 2),
+		Score: int((cpuScore + memoryScore + nvidiaGPUScore) / 2),
```

Comment:
So far without it I think it's ok, but in the future, might have multiple GPU on 1 machine and each GPU could be used by multiple Pods, at this time, I think the nvidiaGPU Score is needed. It should be divided by 3.